p2p/src/net/types.rs

```diff
@@ -142,13 +142,13 @@ pub enum ConnectivityEvent<T: NetworkingService> {
         error: error::P2pError,
     },
 
-    /// Peer misbehaved
-    Misbehaved {
+    /// Protocol violation
+    ProtocolViolation {
```

Comment:
Is there a reason for renaming Misbehaved to ProtocolViolation? I'm not a big fan of this change given that it's well-known what misbehaving means in bitcoin.